File: runtime-sdk/src/types/callformat.rs
//! Types related to call formats.
use crate::core::common::crypto::mrae::deoxysii;

/// Call data key pair ID domain separation context base.
pub const CALL_DATA_KEY_PAIR_ID_CONTEXT_BASE: &[u8] = b"oasis-runtime-sdk/private: tx";

/// A call envelope when using the EncryptedX25519DeoxysII format.
#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
pub struct CallEnvelopeX25519DeoxysII {
    /// Caller's ephemeral public key used for X25519.
    pub pk: [u8; 32],
    /// Nonce.
    pub nonce: [u8; deoxysii::NONCE_SIZE],
    /// Encrypted call data.
    pub data: Vec<u8>,
}

/// A result envelope when using the EncryptedX25519DeoxysII format.
#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
pub struct ResultEnvelopeX25519DeoxysII {
    /// Nonce.
    pub nonce: [u8; deoxysii::NONCE_SIZE],
    /// Encrypted call data.
    pub data: Vec<u8>,
}

File: client-sdk/go/types/callformat.go
package types

import (
	"github.com/oasisprotocol/curve25519-voi/primitives/x25519"
	"github.com/oasisprotocol/deoxysii"
)

// CallEnvelopeX25519DeoxysII is a call envelope when using the EncryptedX25519DeoxysII format.
type CallEnvelopeX25519DeoxysII struct {
	// Pk is the caller's ephemeral public key used for X25519.
	Pk x25519.PublicKey `json:"pk"`
	// Nonce.
	Nonce [deoxysii.NonceSize]byte `json:"nonce"`
	// Epoch is the epoch of the ephemeral runtime key.
	Epoch uint64 `json:"epoch,omitempty"`
	// Data is the encrypted call data.
	Data []byte `json:"data"`
}

// ResultEnvelopeX25519DeoxysII is a result envelope when using the EncryptedX25519DeoxysII format.
type ResultEnvelopeX25519DeoxysII struct {
	// Nonce.
	Nonce [deoxysii.NonceSize]byte `json:"nonce"`
	// Data is the encrypted result data.
	Data []byte `json:"data"`
}
